File: cosc3320/binarySearch.java
import java.io.*;

// Java implementation of iterative Binary Search 
class BinarySearch { 
    // Returns index of x if it is present in arr[], 
    // else return -1 
    int binarySearch(int arr[], int x) 
    { 
        int l = 0, r = arr.length - 1; 
        while (l <= r) { 
            int m = l + (r - l) / 2; 
  
            // Check if x is present at mid 
            if (arr[m] == x) 
                return m; 
  
            // If x greater, ignore left half 
            if (arr[m] < x) 
                l = m + 1; 
  
            // If x is smaller, ignore right half 
            else
                r = m - 1; 
        } 
  
        // if we reach here, then element was 
        // not present 
        return -1; 
    } 
  
    // Driver method to test above 
    public static void main(String args[]) 
    { 
        BinarySearch ob = new BinarySearch(); 
        int m = 2097152;
        int []arr = new int[m]; 
        long startTime = System.currentTimeMillis();
        for (int i = 0; i < 10000000; i++) {
            int result = ob.binarySearch(arr, 2); 
        }
        long endTime = System.currentTimeMillis();
        long totalTime = (endTime - startTime);
        System.out.print("Time elapsed during java search, for m of "); 
        System.out.print(m);
        System.out.print(":");
        System.out.print(totalTime);
        System.out.println("ms");
    } 
} 

File: cosc3320/binarySearch.py
import time

def binarySearch (arr, l, r, x): 
  
    # Check base case 
    if r >= l: 
  
        mid = l + (r - l) // 2
  
        # If element is present at the middle itself 
        if arr[mid] == x: 
            return mid 
          
        # If element is smaller than mid, then it  
        # can only be present in left subarray 
        elif arr[mid] > x: 
            return binarySearch(arr, l, mid-1, x) 
  
        # Else the element can only be present  
        # in right subarray 
        else: 
            return binarySearch(arr, mid + 1, r, x) 
  
    else: 
        # Element is not present in the array 
        return -1
  
# Driver Code 
def main():
    m = 2097152
    arr = [0] * m
  
    # Function call 
    startTime = time.time()
    for i in range(10000000):
        result = binarySearch(arr, 0, len(arr)-1, 2)
        finalTime = time.time() - startTime
    print ("Time elapsed during python3 search, for m of ", m, ": ", finalTime)

main()